Release: 2023.12.0

syuilo opened the pull request:
### Note
- 依存関係の更新に伴い、Node.js 20.10.0が最小要件になりました
- 絵文字の追加辞書を既にインストールしている場合は、お手数ですが再インストールのほどお願いします
- 絵文字ピッカーにピン留め表示する絵文字設定が「リアクション用」と「絵文字入力用」に分かれました。以前の設定は「リアクション用」として使用されます。  

	**影響：**  
	それにより、投稿フォームから表示される絵文字ピッカーのピン留め絵文字がリセットされたように感じるかもしれません（新設された"ピン留め（全般）"の設定が使われるため）。   
	投稿用のピン留め絵文字をアップデート前の状態にするには、以下の手順で操作します。

	1. 「設定」メニューに移動し、「絵文字ピッカー」タブを選択します。
	2. 「ピン留 (全般)」のタブを選択します。
	3. 「リアクション設定から上書きする」ボタンを押すことで、アップデート前の状態に戻すことができます。

### General
- Feat: メールアドレスの認証にverifymail.ioを使えるように (cherry-pick from https://github.com/TeamNijimiss/misskey/commit/971ba07a44550f68d2ba31c62066db2d43a0caed)
- Feat: モデレーターがユーザーのアイコンもしくはバナー画像を未設定状態にできる機能を追加 (cherry-pick from https://github.com/TeamNijimiss/misskey/commit/e0eb5a752f6e5616d6312bb7c9790302f9dbff83)
- Feat: TL上からノートが見えなくなるワードミュートであるハードミュートを追加
- Enhance: 指定したドメインのメールアドレスの登録を弾くことができるように
- Enhance: 公開ロールにアサインされたときに通知が作成されるように
- Enhance: アイコンデコレーションを複数設定できるように
- Enhance: アイコンデコレーションの位置を微調整できるように
- Enhance: つながりの公開範囲をフォロー/フォロワーで個別に設定可能に #12072
- Enhance: ローカリゼーションの更新
- Enhance: 依存関係の更新
- Fix: MFM `$[unixtime ]` に不正な値を入力した際に発生する各種エラーを修正

### Client
- Feat: 今日誕生日のフォロー中のユーザーを一覧表示できるウィジェットを追加
- Feat: 画面に雪を降らせられるように
- Enhance: MFMのアニメーション要素（`tada`, `jelly`, `twitch`, `shake`, `spin`, `jump`, `bounce`, `rainbow`）に `delay` オプションを追加
- Enhance: センシティブと判断されたウェブサイトのサムネイルを非表示に
  - ウェブサイトをセンシティブと判断する仕組みが動いていないため、summalyProxyを使用しないと機能しません。
- Enhance: 投稿フォームの絵文字ピッカーをリアクション時に使用するものと同じのを使用するように #12336 #12560
- Enhance: リアクション用ピン留め絵文字と投稿時の絵文字入力用ピン留め絵文字を分けて設定できるように #12560
- Enhance: 絵文字のオートコンプリート機能強化 #12364
- Enhance: ユーザーのRawデータを表示するページが復活
- Enhance: リアクション選択時に音を鳴らせるように
- Enhance: サウンドにドライブのファイルを使用できるように
- Enhance: ナビゲーションバーに項目「キャッシュを削除」を追加
- Enhance: Shareページで投稿を完了すると、親ウィンドウ（親フレーム）にpostMessageするように
- Enhance: チャンネル、クリップ、ページ、Play、ギャラリーにURLのコピーボタンを設置 #11305
- Enhance: ノートプレビューに「内容を隠す」が反映されるように
- Enhance: データセーバーでコードハイライトの読み込みを削減できるように
- Enhance: データセーバーの適用範囲を個別で設定できるように
	- 従来のデータセーバーの設定はリセットされます
- Enhance: タイムライン上のタブからリスト、アンテナ、チャンネルの管理ページにジャンプできるように
- Enhance: ユーザー名、プロフィール、お知らせ、ページの編集画面でMFMや絵文字のオートコンプリートが使用できるように
- Enhance: プロフィール、お知らせの編集画面でMFMのプレビューを表示できるように
- Enhance: 絵文字の詳細ページに記載される情報を追加
- Enhance: リアクションの表示幅制限を設定可能に
- Enhance: Unicode 15.0のサポート
- Enhance: コードブロックのハイライト機能を利用するには言語を明示的に指定させるように
	- MFMでコードブロックを利用する際に意図しないハイライトが起こらないようになりました
	- 逆に、MFMでコードハイライトを利用したい際は言語を明示的に指定する必要があります  
	（例: ` ```js ` → Javascript, ` ```ais ` → AiScript）
-	Enhance: 絵文字などのオートコンプリートでShift+Tabを押すと前の候補を選択できるように
- Enhance: チャンネルに新規の投稿がある場合にバッジを表示させる
- Enhance: サウンド設定に「サウンドを出力しない」と「Misskeyがアクティブな時のみサウンドを出力する」を追加
- Enhance: 設定したタグをトレンドに表示させないようにする項目を管理画面で設定できるように
- Enhance: 絵文字ピッカーのカテゴリに「/」を入れることでフォルダ分け表示できるように
- Fix: 「設定のバックアップ」で一部の項目がバックアップに含まれていなかった問題を修正
- Fix: ウィジェットのジョブキューにて音声の発音方法変更に追従できていなかったのを修正 #12367
- Fix: コードエディタが正しく表示されない問題を修正
- Fix: プロフィールの「ファイル」にセンシティブな画像がある際のデザインを修正
- Fix: 一度に大量の通知が入った際に通知音が音割れする問題を修正
- Fix: 共有機能をサポートしていないブラウザの場合は共有ボタンを非表示にする #11305
- Fix: 通知のグルーピング設定を変更してもリロードされるまで表示が変わらない問題を修正 #12470
- Fix: 長い名前のチャンネルにおける投稿フォームの表示が崩れる問題を修正
- Fix: セキュリティ向上のためAiScriptの`Mk:apiExternal`を無効化
- Fix: ノート中の絵文字をタップして「リアクションする」からリアクションした際にリアクションサウンドが鳴らない不具合を修正
- Fix: ノート中のリアクションの表示を微調整 #12650
- Fix: AiScriptの`readline`が不正な値を返すことがある問題を修正
- Fix: 投票のみ/画像のみの引用RNが、通知欄でただのRNとして判定されるバグを修正
- Fix: CWをつけて引用RNしても、普通のRNとして扱われてしまうバグを修正しました。
- Fix: 「画像が1枚のみのメディアリストの高さ」を「デフォルト」以外に設定していると、CWの中などに添付された画像が見られないバグを修正
- Fix: DeepL TranslationのPro accountトグルスイッチが表示されていなかったのを修正
- Fix: twitterの埋め込みカード内リンクからリンク先を開けない問題を修正
- Fix: WebKitブラウザー上でも「デバイスの画面を常にオンにする」機能が効くように
- Fix: ページ一覧ページの表示がモバイル環境において崩れているのを修正
- Fix: MFMでルビの中のテキストがnyaizeされない問題を修正

### Server
- Enhance: MFM `$[ruby ]` が他ソフトウェアと連合されるように
- Enhance: Meilisearchを有効にした検索で、ユーザーのミュートやブロックを考慮するように
- Enhance: カスタム絵文字のインポート時の動作を改善
- Fix: 時間経過により無効化されたアンテナを再有効化したとき、サーバ再起動までその状況が反映されないのを修正 #12303
- Fix: ロールタイムラインが保存されない問題を修正
- Fix: api.jsonの生成ロジックを改善 #12402
- Fix: 招待コードが使い回せる問題を修正
- Fix: 特定の条件下でチャンネルやユーザーのノート一覧に最新のノートが表示されなくなる問題を修正
- Fix: 何もノートしていないユーザーのフィードにアクセスするとエラーになる問題を修正
- Fix: リストタイムラインにてミュートが機能しないケースがある問題と、チャンネル投稿がストリーミングで流れてきてしまう問題を修正 #10443
- Fix: 「みつける」のなかにミュートしたユーザが現れてしまう問題を修正 #12383
- Fix: Social/Local/Home Timelineにてインスタンスミュートが効かない問題
- Fix: ユーザのノート一覧にてインスタンスミュートが効かない問題
- Fix: チャンネルのノート一覧にてインスタンスミュートが効かない問題
- Fix: 「みつける」が年越し時に壊れる問題を修正
- Fix: アカウントをブロックした際に、自身のユーザーのページでノートが相手に表示される問題を修正
- Fix: モデレーションログがモデレーターは閲覧できないように修正
- Fix: ハッシュタグのトレンド除外設定が即時に効果を持つように修正
- Fix: HTTP Digestヘッダのアルゴリズム部分に大文字の"SHA-256"しか使えない
- Fix: 管理者用APIのアクセス権限が適切に設定されていない問題を修正

samunohito committed:
DeepL TranslationのPro accountトグルスイッチが表示されていなかったのを修正 (#12355)

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

nafu-at committed:
feat(moderation): モデレーターがユーザーのアイコンもしくはバナー画像を未設定状態にできる機能を追加 (MisskeyIO#222)

Co-authored-by: まっちゃとーにゅ <17376330+u1-liquid@users.noreply.github.com>

nafu-at committed:
メールアドレスの認証にverifymail.ioを使えるようにする。

syuilo committed:
Update CHANGELOG.md

syuilo committed:
lint fix

syuilo committed:
Update CHANGELOG.md

syuilo committed:
tweak of 2b6f789a5b

lynlynlynx committed:
ページ一覧ページの表示がモバイル環境において崩れているのを修正 (#12354)

* fix style of list of pages on mobile

* overflow clip に変えた

hideki0403 committed:
fix: 「設定のバックアップ」に一部の設定項目が含まれていない問題を修正 (#12366)

* fix: 一部の設定項目がバックアップに含まれていなかったのを修正

* update: CHANGELOG.md

* remove: バックアップ不要な項目を削除

fruitriin committed:
backendのプロジェクトで単体で start できないのを修正 (#12371)

syuilo committed:
fix(backend): ロールタイムラインが保存されない問題を修正

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

anatawa12 committed:
update api.md (#12379)

for API changes in b65fd349812fe3c89b5face6ec5c12823459d7df

samunohito committed:
json-schema配下の最新化 (#12312)

* user.ts、page.ts、drive-folder.tsを各EntityServiceの戻り値をもとに最新化

* 再確認

* fix error

* note以外の残りのファイルを対応

* fix CHANGELOG.md

* fix CHANGELOG.md

* fix user.ts

* fix user.ts

* コメント対応

* fix note.ts

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

syuilo committed:
update typescript to 5.3

samunohito committed:
note.tsのchannelを正しい形にしたことにより表出化した型チェックエラーを修正 (#12395)

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

samunohito committed:
サーバ起動時にアンテナが非アクティブだった場合、アクティブ化しても再起動するまで反映されない (#12391)

* サーバ起動時にアンテナが非アクティブだった場合、アクティブ化しても再起動するまで反映されない

* Fix CHANGELOG.md

* lastUsedAtの更新に不備が出るので修正

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

nenohi committed:
広告掲載ページにてfilterをわかりやすく (#12385)

samunohito committed:
サウンド再生方法の変更に追従できていなかった所を修正 (#12368)

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

nafu-at committed:
fix: verifymail.io APIの設定項目が反映されない (#12399)

eai04191 committed:
twitter埋め込みのsandbox属性にallow-popups-to-escape-sandboxを追加 (#12400)

Co-authored-by: unarist <m.unarist@gmail.com>

tai-cha committed:
taichanne30 -> tai-cha (#12405)

samunohito committed:
swagger-cli validateがvalidとなるapi.jsonを作れるようにする (#12403)

* api.jsonがswagger-cli validateでエラーにならないように生成ロジックを修正

* フィールドの消し方に不備があったので変更

* バックエンドを起動しなくてもapi.jsonを作れるようにした

* deepCopyしてからレスポンス部分を作るようにした

* fix CHANGELOG.md

* securitySchemesの定義を復活＆ApiCallServiceの実装的にベアラトークンなのでその形で

* bodyが無い（空オブジェクト）のときはrequestBodyを描画しないようにする

* allowGetがtrueな項目はget用の記載も作成

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>
Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
fix lint, resolve code smell

Co-Authored-By: まっちゃとーにゅ <17376330+u1-liquid@users.noreply.github.com>

samunohito committed:
絵文字のオートコンプリート強化の対応 (#12365)

* 前方一致・部分一致でなくても近似値でヒットするように

* fix CHANGELOG.md

* for of に変更

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

anatawa12 committed:
Hard mute (#12376)

* feat(backend,misskey-js): hard mute storage in backend

* fix(backend,misskey-js): mute word record type

* chore(frontend): generalize XWordMute

* feat(frontend): configure hard mute

* feat(frontend): hard mute notes on the timelines

* lint(backend,frontend): fix lint failure

* chore(misskey-js): update api.md

* fix(backend): test failure

* chore(frontend): check word mute for reply

* chore: limit hard mute count

kakkokari-gtyih committed:
fix(backend): 招待コードが使い回せる問題を修正 (#12423)

* (fix) 招待コードを一度のみ利用できるように

* Update Changelog

kakkokari-gtyih committed:
enhance/fix(AP/frontend): 最近追加されたMFMのもろもろを修正 (#12420)

* (enhance) MFM rubyが連合されるように

* Update Changelog

* Update Changelog

* (fix) unixtimeのフォールバック (AP)

* (fix) unixtimeのフォールバック (frontend)

* Update Changelog

anatawa12 committed:
Use generate-api-json for api.json diff GitHub comment (#12408)

* ci: use generate-api-json to get api.json changes

* restore copying default.yml

* refactor: get api.json with single workflow

* ci: api.jsonのdiffをbackendが変更されたときのみ取るように

kakkokari-gtyih committed:
fix(frontend): コードエディタが正しく表示されない問題を修正 (#12418)

* (fix) コードエディタが正しく表示されない問題を修正

* Update Changelog

kakkokari-gtyih committed:
fix(frontend): プロフィールの「ファイル」にセンシティブな画像がある際のデザインを修正 (#12424)

* (fix) 招待コードを一度のみ利用できるように

* Update Changelog

* (fix) profile media grid

* Update Changelog

* Change Changelog

u1-liquid committed:
enhance(frontend): 絵文字のオートコンプリートのアルゴリズムの改善 (MisskeyIO#261)

* 実際は同じ絵文字なら重複してサジェストに出ないように
* エイリアスではない絵文字>前方一致>部分一致>あいまい検索順で表示されるようになるように

mappi-pr committed:
feat 12325 (#12425)

camilla-ett committed:
feat(frontend): リアクションの横幅を150pxに制限するかどうかユーザーが選べるように (#12416)

* feat(frontend): リアクションの横幅を150pxに制限するかどうかユーザーが選べるように

* localesの変更をjs-JP.ymlのみに修正し、日本語をより分かりやすく

* クラス名を.icon から .limitWidthに変更

syuilo committed:
Update index.d.ts

syuilo committed:
update node to 20.10.0

woxtu committed:
Replace deprecated `Repository.findOneById()` (#12426)

anatawa12 committed:
fix: hard mute limit not applied (#12428)

* fix: hard mute limit not applied

* Update CHANGELOG.md

camilla-ett committed:
Feat(frontend): コントロールパネル「通報」において、通報者もリンクで飛べるように修正 (#12427)

hideki0403 committed:
fix: 特定の条件下でチャンネルやユーザーのノート一覧に最新のノートが表示されなくなる問題を修正 (#12431)

* fix: 特定の条件下でチャンネルやユーザーのノート一覧に最新のノートが表示されなくなる問題を修正

* update: CHANGELOG.md

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

kakkokari-gtyih committed:
enhance(frontend): ユーザーのRawデータを読めるページを復活させる (#12436)

* (add) User raw page

* Update Changelog

* fix lint

woxtu committed:
Add mocks for Web Audio API (#12457)

zyoshoka committed:
fix(backend): 何もノートしていないユーザーのフィードにアクセスするとエラーになる問題を修正 (#12455)

* fix(backend): 何もノートしていないユーザーのフィードにアクセスするとエラーになる問題を修正

* Update CHANGELOG.md

* add test

* fix: incorrect bob's username

zyoshoka committed:
fix(backend): rename FunoutTimelineService to FanoutTimelineService (#12453)

zyoshoka committed:
fix(backend): ギャラリーの人気の投稿の選出にidを用いるように (#12448)

anatawa12 committed:
fix: query error in notes/featured (#12439)

kakkokari-gtyih committed:
enhance(frontend): リアクション選択時に音を流せるように (#12441)

* (add) リアクション選択時に音を鳴らせるように

* Update Changelog

* tweak sound

* tweak sound

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

kakkokari-gtyih committed:
fix(frontend): 通知音がほぼ同時に鳴った場合は再生をブロックするように（音割れ防止） (#12433)

* (fix) 通知音がダブって音割れしないように

* Update Changelog

acid-chicken committed:
chore: create AudioContext when it is needed (#12460)

samunohito committed:
サウンド設定に「サウンドを出力しない」と「Misskeyがアクティブな時のみサウンドを出力する」を追加 (#12342)

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

acid-chicken committed:
chore(frontend/MkMediaVideo): loop and autoplay silent videos (#12392)

fumons committed:
fix: wake lock error in safari etc (#12464)

woxtu committed:
Fix a frontend testing script (#12471)

anatawa12 committed:
fix: error can be happened if animation is on and hard mute matches (#12474)

anatawa12 committed:
style: fix lint error of 6acaded8 (#12476)

kakkokari-gtyih committed:
enhance(frontend): 通知音にドライブのファイルを使用できるように (#12447)

* (enhance) サウンドにドライブのファイルを使用できるように

* Update Changelog

* fix

* fix design

* fix design

* Update store.ts

* (fix) ファイル名表示

* refactor

* (refactor) better types

* operationTypeとsoundTypeの混同を防止

* (refactor)

* (fix)

* enhance jsdoc

* driveFile -> _driveFile_

syuilo committed:
update deps

syuilo committed:
2023.12.0-beta.1

KisaragiEffective committed:
docs(changelog): GHSA-3f39-6537-3cgc について追記 (#12482)

kakkokari-gtyih committed:
fix(docs): "docs(changelog): GHSA-3f39-6537-3cgc について追記 (#12482)" (#12483)

* Revert "docs(changelog): GHSA-3f39-6537-3cgc について追記 (#12482)"

This reverts commit d58ec4e65ba2336d7dc06f767d1dcaf43d29e1e2.

* Update CHANGELOG.md

samunohito committed:
通知グルーピング設定の即時反映対応 (#12485)

* wip

* ログ出しの削除

* fix CHANGELOG.md

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>

yupix committed:
fix: packedNoteSchemaにclippedCountが抜けている (#12499)

anatawa12 committed:
fix: 音声が一切鳴らなくなる可能性がある (#12491)

* chore: 音声が一切鳴らなくなる可能性を軽減

https://github.com/misskey-dev/misskey/pull/12433#discussion_r1405774767

* chore: IIFEではなくPromise.prototype.finallyを使用するように

zyoshoka committed:
fix(frontend): WebKitブラウザー上でも「デバイスの画面を常にオンにする」機能が効くように (#12484)

* fix(frontend): WebKitブラウザー上でもkeepScreenOnが効くように

* chore: add comment

woxtu committed:
Update the Vitest configuration (#12493)

anatawa12 committed:
ci: use refs/pull/*/merge to get head (#12508)

kakkokari-gtyih committed:
enhance(frontend): Shareページでの投稿完了時にpostMessageを発火するように (#12505)

* enhance(frontend): Shareページでの投稿完了時にpostMessageを発火

* Update Changelog

* fix

* 名前の混同をさける

* 名前をわかりやすくする

* watchを使わずパフォーマンス改善

samunohito committed:
Fix: navigator.share未サポートの場合は共有ボタンを非表示にする（＋URLのコピーボタンを設置） (#12506)

* navigator.share未サポートの場合は共有ボタンを非表示にする

* fix CHANGELOG.md

* ライセンス表示追加

* URLのコピーボタンを設置

zyoshoka committed:
enhance(frontend): ノートプレビューにCWが反映されるように (#12509)

* enhance(frontend): ノートプレビューにCWが反映されるように

* Update CHANGELOG.md

* refactor: 不要な条件を除去

* Revert "refactor: 不要な条件を除去"

This reverts commit e4eff689bd1f2df78411a2f01ee05434e2d298a8.

* fix: やっぱり不要な条件だった

GrapeApple0 committed:
enhance: 設定したタグをトレンドに表示させないようにする項目を管理画面で設定できるように (#12512)

* enhance: hiddenTagsを管理画面で設定できるように

* Update locales/ja-JP.yml

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

KisaragiEffective committed:
refactor(frontend): give local variable to explicit type annotation to avoid TS7043 (#12495)

* refactor: give local variable to explicit type annotation to avoid TS7043

* chore: fix lint error

Hoto-Cocoa committed:
fix(docker): cannot build docker image on some environments (#12494)

hideki0403 committed:
enhance(dev): 開発モード時にlocaleと型定義が自動的に再生成されるように (#12481)

* enhance: localeを任意のタイミングでリビルドできるように

* enhance: localeも監視し、必要であればlocaleをリビルドするように

* feat: devモードの時のみナビゲーションバーからキャッシュクリアができるように

* refactor: キャッシュクリア部分を共通化

* fix: localesのファイル変更イベントが取れないのを修正

* fix: replaceAllでコケるのを修正

* change: 開発モードに関係なくナビゲーションバーからキャッシュクリアできるように

* refactor: 必要のないリビルドをしないように

* update: CHANGELOG.md

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

kakkokari-gtyih committed:
feat(frontend): 今日誕生日のフォロー中のユーザーを一覧表示できるウィジェットを追加 (#12450)

* (add) 今日誕生日のフォロイー一覧表示

* Update Changelog

* Update Changelog

* 実装漏れ

* create index

* (fix) index

Srgr0 committed:
絵文字詳細ページに記載する情報を追加 (#12417)

* Update emojis.emoji.vue

* Update CHANGELOG.md

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

yupix committed:
fix: invite系の戻り値が間違っている close #12517 (#12518)

Commit:
rename docker-compose.yml.example to docker-compose_example.yml (#12530)

* rename docker-compose.yml.example to docker-compose_example.yml

* fix: dockle.yml

qwreey committed:
Fix: missing receiver warn is not disappear (#12538)

samunohito committed:
fix #12528 (#12536)

zyoshoka committed:
fix(frontend): 長い名前のチャンネルにおける投稿フォームの表示が崩れる問題を修正 (#12524)

* fix(frontend): 長い名前のチャンネルにおける投稿フォームの表示が崩れる問題を修正

* Update CHANGELOG.md

anatawa12 committed:
chore: フォローしたときにHTLをパージしなくする (#12522)

kakkokari-gtyih committed:
(fix) 翻訳のダブりを解消 (#12519)

anatawa12 committed:
chore: remove unimplemented excludeNsfw (#12520)

meronmks committed:
feat(frontend): 絵文字ピッカーのカテゴリを多階層フォルダで分類できるように (#12132)

* Update CHANGELOG.md

* Feat:emoji picker folder select

* Fix: lint error

* Fix: lint error 2

* Fix: lint error 3

* カスタム絵文字のカテゴリ表示部分が長かったので短くした

* エフェクトが壊れて出ないのを修正

* padding 18px -> 9px

* Update CHANGELOG.md

* Revert: en-US.yml

* chg: Folder -> folder

* chg: isChildrenExits -> isChildren

* chg: isChildren -> categoryFolderFlag

* カテゴリ末尾が / の場合ピッカーから消失するので「その他」と扱い対応

* 特定のパターンのカテゴリ名でピッカーに出てこないのを修正

「i18n.ts.other」や「/」始まりの場合壊れる

* chg: categoryFolderFlag -> hasChildSection

* code format

* Del: ti-fw

* fix

* 絵文字とフォルダの表示順序入れ替え

* ネストした場合にパネルでどこまでがどのフォルダのものかをわかりやすくした

* fix lint

* カテゴリの名前が長いと表示がおかしくなる問題を修正

Co-authored-by: まっちゃとーにゅ <17376330+u1-liquid@users.noreply.github.com>

---------

Co-authored-by: tamaina <tamaina@hotmail.co.jp>
Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>
Co-authored-by: atsuchan <83960488+atsu1125@users.noreply.github.com>
Co-authored-by: Masaya Suzuki <15100604+massongit@users.noreply.github.com>
Co-authored-by: Kagami Sascha Rosylight <saschanaz@outlook.com>
Co-authored-by: taiy <53635909+taiyme@users.noreply.github.com>
Co-authored-by: xianon <xianon@hotmail.co.jp>
Co-authored-by: kabo2468 <28654659+kabo2468@users.noreply.github.com>
Co-authored-by: YS <47836716+yszkst@users.noreply.github.com>
Co-authored-by: Khsmty <me@khsmty.com>
Co-authored-by: Soni L <EnderMoneyMod@gmail.com>
Co-authored-by: mei23 <m@m544.net>
Co-authored-by: daima3629 <52790780+daima3629@users.noreply.github.com>
Co-authored-by: Windymelt <1113940+windymelt@users.noreply.github.com>
Co-authored-by: Ebise Lutica <7106976+EbiseLutica@users.noreply.github.com>
Co-authored-by: nenohi <kimutipartylove@gmail.com>
Co-authored-by: Acid Chicken (硫酸鶏) <root@acid-chicken.com>
Co-authored-by: rinsuki <428rinsuki+git@gmail.com>
Co-authored-by: FineArchs <133759614+FineArchs@users.noreply.github.com>
Co-authored-by: まっちゃとーにゅ <17376330+u1-liquid@users.noreply.github.com>

camilla-ett committed:
fix(backend): カスタム絵文字のインポート時の動作を修正 (#12360)

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

paihu committed:
fix(frontend): MFM ruby nyaize (#12362)

anatawa12 committed:
Refine fanout timeline (#12507)

* chore(endpoints/hybrid-timeline): don't pack inside getFromDb

* chore(endpoints/hybrid-timeline): Redisから取得する部分のうちSTLに依存しなそうなところを別のServiceに切り出し

* chore(endpoints/local-timeline): FanoutTimelineEndpointServiceで再実装

* chore(endpoints/channels/timeline): FanoutTimelineEndpointServiceで再実装

* chore(endpoints/timeline): FanoutTimelineEndpointServiceで再実装

* chore(endpoints/user-list-timeline): FanoutTimelineEndpointServiceで再実装

* chore(endpoints/users/notes): FanoutTimelineEndpointServiceで再実装

* chore: add useDbFallback to FanoutTimelineEndpointService.timeline and always true for channel / user note list

* style: fix lint error

* chore: split logic to multiple functions

* chore: implement redis fallback

* chore: 成功率を上げる

* fix: db fallback not working

* feat: allowPartial

* chore(frontend): set allowPartial

* chore(backend): remove fallbackIfEmpty

HTL will never be purged so it's no longer required

* fix: missing allowPartial in channel timeline

* fix: type of timelineConfig in hybrid-timeline

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

mei23 committed:
fix: Filter featured collection (#12541)

samunohito committed:
fix: #12544 (#12545)

* meを渡し忘れている

* fix CHANGELOG.md

* Revert "fix CHANGELOG.md"

This reverts commit aaee4e9b8a6abf510f393bc02282f6ac016d2124.

samunohito committed:
バックエンドが生成するapi.jsonからmisskey-jsの型を作成する (#12434)

* ひとまず生成できるところまで

* ファイル構成整理

* 生成コマンド整理

* misskey-jsへの組み込み

* fix generator.ts

* wip

* fix generator.ts

* fix package.json

* 生成ロジックの調整

* 型レベルでのswitch-case機構をmisskey-jsからfrontendに持ち込めるようにした

* 型チェック用のtsconfig.jsonを作成

* 他のエンドポイントを呼ぶ関数にも適用

* 未使用エンティティなどを削除

* misskey-js側で手動定義されていた型を自動生成された型に移行（ただしapi.jsonがvalidでなくなってしまったので後で修正する）

* messagingは廃止されている（テストのビルドエラー解消）

* validなapi.jsonを出力できるように修正

* 修正漏れ対応

* Ajvに怒られて起動できなかったところを修正

* fix ci（途中）

* パラメータenumをやめる

* add command

* add api.json

* 都度自動生成をやめる

* 一気通貫スクリプト修正

* fix ci

* 生成ロジック修正

* フロントの型チェックは結局やらなかったので戻しておく

* fix pnpm-lock.yaml

* add README.md

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>
Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

samunohito committed:
Revert "fix #12528 (#12536)" (#12548)

This reverts commit a5f0b5ec74940b0c53242dfc64c322139c91e362.

shiosyakeyakini-info committed:
fix: withChannelNotesとwithFilesを同時に指定したときの考慮 (#12550)

Co-authored-by: sorairo <sorairo@shiosyakeyakini.info>

samunohito committed:
リストタイムラインでミュートが貫通してしまう問題に対処 (#12534)

* ユーザリストTL系の各種動作を修正・統一

* fix

* fix CHANGELOG.md

* テスト追加

7ka-Hiira committed:
fix(backend): /emojiにおける拡張子の削除方法を修正 (#12543)

Co-authored-by: Acid Chicken (硫酸鶏) <root@acid-chicken.com>

kakkokari-gtyih committed:
enhance/feat(frontend): データセーバーの改良・強化 (#12526)

* enhance(frontend): データセーバーを個別で設定できるように

* Update Changelog

* fix design

* (fix) 設定が当たらない

* fix test（無理やり感）

* (fix) 設定がない状態ですべて有効・向操作が効かない

* fix

* tweak

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

fruitriin committed:
may be fix ruby justify on safari (#12551)

samunohito committed:
enhance(misskey-js) misskey-jsのストリーミングAPI定義をバックエンドに追従 (#12552)

* (enhance) misskey-jsのストリーミングAPI定義をバックエンドに追従

* fix ci

* fix ci

6543 committed:
fix(backend): enhance nodeinfo by export instance admin via nodeAdmins key (#12503)

https://codeberg.org/thefederationinfo/nodeinfo_extension

acid-chicken committed:
fix(backend): reject malformed timestamp (#12554)

samunohito committed:
入力フォームでもリアクション選択時に使用するピッカーを使うようにしたい (#12337)

* 入力フォームでもリアクション選択時に使用するピッカーを使うようにしたい

* erase console.log

* fix CHANGELOG.md

* reaction-picker.ts を戻し、今回の対応を入れた emoji-picker.ts を新たに作成

* fix CHANGELOG.md

* tweak

---------

Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>
Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

samunohito committed:
fix (backend): 「みつける」のなかにミュートしたユーザが現れてしまう問題を修正 (#12559)

* fix (backend): 「みつける」のなかにミュートしたユーザが現れてしまう問題を修正

* fix

anatawa12 commented:
CI落ちは #12408 の影響ですね。すみません。 headに`generate-api-json`がないのを想定していませんでした。一旦revert等、なにか対応したほうがいいでしょうか？

samunohito committed:
fix dev build (#12566)

anatawa12 committed:
Filter User / Instance Mutes in FanoutTimelineEndpointService (#12565)

* fix: unnecessary logging in FanoutTimelineEndpointService

* chore: TimelineOptions

* chore: add FanoutTimelineName type

* chore: forbid specifying both withReplies and withFiles since it's not implemented correctly

* chore: filter mutes, replies, renotes, files in FanoutTimelineEndpointService

* revert unintended changes

* use isReply in NoteCreateService

* fix: excludePureRenotes is not implemented

* fix: replies to me is excluded from local timeline

* chore(frontend): forbid enabling both withReplies and withFiles

* docs(changelog): インスタンスミュートが効かない問題の修正について言及

yupix committed:
feat: ユーザースキーマの改善 (#12568)

* chore: notifyにenumを設定

* feat: securityKeysListの型を明確に

* feat: notificationRecieveConfigにpropertiesを定義

* chore: misskey.jsのmodelを更新

* fix: as constをつけ忘れている

anatawa12 committed:
chore: 自分へのリプライのみ走査するように (#12570)

samunohito committed:
fix (frontend): 絵文字ピッカー経由で投稿欄に絵文字を入れた際、ソフトウェアキーボードが立ち上がらないようにする (#12561)

syuilo committed:
update deps

syuilo committed:
2023.12.0-beta.2

anatawa12 committed:
fix(backend): エポックを固定することで年越し時にトレンドが壊れる問題を修正 (#12567)

tamaina commented:
メニュー内のトグルが操作できない

yuriha-chan committed:
タイムラインの「リノートを表示」のトグルスイッチが反応しない問題を直す (#12577)

* [frontend] Fix renote toggle switch

* Fix MkMenu rather than usage

anatawa12 committed:
chore: hide thumbnail if website is sensitive (#12581)

yupix committed:
feat: Roleに関するSchemaを追加 (#12572)

* feat: Roleに関連するschemaを追加

* feat: 新しいRoleSchemaを使うように

* chore: misskey.jsのデータを更新

* chore: misskey-js.api.mdを更新

zyoshoka committed:
refactor(frontend): 非推奨となったReactivity Transformを使わないように (#12539)

* refactor(frontend): 非推奨となったReactivity Transformを使わないように

* refactor: 不要な括弧を除去

* fix: 不要なアノテーションを除去

* fix: Refの配列をrefしている部分の対応

* refactor: 不要な括弧を除去

* fix: lint

* refactor: Ref、ShallowRef、ComputedRefの変数の宣言をletからconstに置換

* fix: type error

* chore: drop reactivity transform from eslint configuration

* refactor: remove unnecessary import

* fix: 対応漏れ

ryanho committed:
chore: Add descriptions for "MeiliSearch" and "allowedPrivateNetworks" to example.yml (#12594)

* Update example.yml, add descriptions for some items

Add descriptions for "MeiliSearch" and "allowedPrivateNetworks"

* Update docker_example.yml

Add descriptions for "MeiliSearch" and "allowedPrivateNetworks"

anatawa12 committed:
fix(backend): pagination with sinceId broken (#12586)

* fix(backend): pagination with sinceId broken

* fix(backend): pagination with sinceId broken for dbFallback

kanarikanaru committed:
enhance: meilisearchを有効にしてもミュートやブロックを考慮するように (#12575)

* enhance: meilisearchを有効にしてもミュートやブロックを考慮するように

* Update CHANGELOG.md

kanarikanaru committed:
fix(backend): ブロックした相手から自分のノートが見えないように(/users/featured-notes, /users/notes) (#12511)

* fix: ブロックした相手から自分のノートが見えないように(ユーザー,チャンネル)

* Update CHANGELOG.md

* /users/featured-notesでもブロックを考慮するように

* cacheServiceを使うように

* /channels/timeline.tsで必要のないnoteFilterを持たないように

* Update CHANGELOG.md

* FanoutTimelineEndpointServiceへの対応

- ブロックされている場合は、/users/notesでノートが表示されない
- ミュートしている場合は、ノートが表示される

samunohito committed:
chore: 開発モードでフロントエンドとバックエンドを独立して起動するようにする（再） (#12593)

* [wip]run standalone vite

* [wip]run standalone vite

* some fix (tabler icons, sw, streaming)

* fix theme

* fix run scripts

* favicon

* client-assets

* cssの読み込み順序とCSP設定の変更

* fix lang change

* fix clientManifest

* baseを相対パスにしてドメイン直下とサブディレクトリ配下両方に対応

* 色々修正

* 色々修正

* 色々修正

* fix

* Revert "client-assets"

This reverts commit 582601e90eb771875bdf8aba263da2316a59d01b.

# Conflicts:
#	packages/frontend/vite.config.ts

* 色々修正

* fix

* fix

* add url and proxy to server proxy

* Update packages/frontend/src/index.html

* wip

* Merge remote-tracking branch 'origin/develop' into feat/launch-standalone-frontend

# Conflicts:
#	packages/frontend/src/pages/welcome.entrance.a.vue

* Merge remote-tracking branch 'origin/develop' into feat/launch-standalone-frontend

# Conflicts:
#	packages/frontend/src/pages/welcome.entrance.a.vue

* fix tabler load

* Apply suggestions from code review

* Update packages/frontend/src/index.html

* fix

* fix vite.config.local-dev.ts

* fix CONTRIBUTING.md

---------

Co-authored-by: FruitRiin <nassii74@gmail.com>
Co-authored-by: osamu <46447427+sam-osamu@users.noreply.github.com>
Co-authored-by: 果物リン <fruitriin@riinswork.space>
Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>
Co-authored-by: ozelot <contact@ozelot.dev>

samunohito committed:
fix CONTRIBUTING.md (#12600)

dependabot[bot] committed:
chore(deps): bump actions/labeler from 4 to 5 (#12584)

Bumps [actions/labeler](https://github.com/actions/labeler) from 4 to 5.
- [Release notes](https://github.com/actions/labeler/releases)
- [Commits](https://github.com/actions/labeler/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/labeler
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kakkokari-gtyih committed:
enhance(frontend): ウィジェットを非表示にできるPageMetaを追加 (#12456)

* (enhance) ウィジェットを非表示にできるPageMetaを追加

* fix lint

* rename

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

tkmrgit commented:
>Fix: リストタイムラインにてミュートが機能しないケースがある問題と、チャンネル投稿がストリーミングで流れてきてしまう問題を修正

がもし #11921 の修正を含むものである場合はそちらを`close`してください

ikasoba committed:
fix: `secure: true` なエンドポイントの型が misskey-js に含まれていない (#12603)

* 作った

* 修正

* 修正

kakkokari-gtyih committed:
fix(dev) 一部のアセットが読み込まれない問題を修正

samunohito committed:
Merge pull request #12604 from kakkokari-gtyih/fix-dev-0

fix(dev-frontend) 一部のアセットが読み込まれない問題を修正

kakkokari-gtyih committed:
fix(dev-frontend) サーバーサイドのHTMLと噛み合わない部分を修正 (#12605)

u1-liquid committed:
enhance(frontend): window.openやaタグにnoopenerオプションをつける (MisskeyIO#283)

zyoshoka committed:
chore: fix labeler's config (#12609)

kakkokari-gtyih committed:
fix(dev-frontend) 足りてないCSPを追加 (#12606)

* fix(dev-frontend) サーバーサイドのHTMLと噛み合わない部分を修正

* cspをなおした

* typo

zyoshoka committed:
chore: labelerが治っていなかったのを修正 (#12610)

* fix ci

* fix

* fix labeler.yml

* Revert "fix labeler.yml"

This reverts commit 9b6a7d02cdd65ab9ad14c666c6ebe4eadeabeadc.

---------

Co-authored-by: samunohito <46447427+samunohito@users.noreply.github.com>

syuilo committed:
fix(frontend): disable Mk:apiExternal

syuilo committed:
update deps

samunohito committed:
Fix/vue import error on intellij (#12612)

* Fix fix labeler config (#8)

* fix vue import error

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

yuriha-chan committed:
Misskey Playのノート投稿画面で「内容を隠す」を設定できるようにする (#12576)

* Add the content warning option in AiScript UI postFormButton

* Fix initial CW in postFormButton

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
2023.12.0-beta.3

samunohito committed:
(dev) index.htmlにmeta[name=viewport]がなかったので追加＆足りてないCSP追加 (#12613)

* (dev) index.htmlにmeta[name=viewport]がなかったので追加＆足りてないCSP追加

* fix tab

woxtu committed:
Fix indentation (#12615)

acid-chicken committed:
refactor(frontend): remove redundant class names (#12618)

syuilo commented:
https://github.com/misskey-dev/misskey/pull/12617 マージしたくらいの段階でリリースしたい

samunohito commented:
明日～明後日くらいにはDraftからOpenにできると思います

woxtu committed:
Fix trailing commas (#12628)

syuilo committed:
refactor(frontend): menuのdividerをnullで表現するのをやめる

syuilo committed:
fix type

1Step621 committed:
Enhance(frontend): リスト/アンテナ/チャンネルをタイムラインから新規作成できるように (#12629)

* add short leads to lists, antennas, and channels

* remove unused import

* add CHANGELOG.md

* hide separator when there is no item

* fix mistakes

* Update timeline.vue

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo commented:
あーデコレーション複数設定もやっちゃうか

tassoman committed:
adding color-scheme light to WidgetAichan (#12638)

camilla-ett committed:
Fix(backend): inboxJobPerSecのデフォルト値を16から32に (#12631)

YAVIIGI committed:
fix(frontend): ノート中の絵文字をタップして「リアクションする」を押したときにリアクションサウンドが鳴るようにする (#12624)

* Add sound.play() in copy reaction

* Update CHANGELOG.md

* fix lint error

syuilo committed:
enhance: アイコンデコレーションを複数設定できるように

kakkokari-gtyih committed:
fix(frontend): MkAnimBgをリサイズに対応させる (#12642)

* (fix) MkAnimBgをリサイズに対応させる

* fix lint

* refactor

syuilo committed:
fix(client): fix glitch when attach/detach avatar decoration

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

syuilo committed:
fix(backend): モデレーションログがモデレーターは閲覧できないように修正

Fix #12622

samunohito committed:
(fix) デフォルト表示時のヘッダにあるチャンネルボタンが反応しない現象の修正 (#12648)

* dividerの仕変に追従

* fix type

syuilo committed:
enhance(frontend): 同じ種類のデコレーションを複数付けられるように

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

syuilo committed:
fix swcrc

samunohito committed:
refactor(frontend) $i の型情報にtokenを追加 (#12649)

1Step621 committed:
Enhance(frontend): MFMや絵文字が使える入力ボックスでオートコンプリートを使えるように (#12643)

* rich autocomplete for use in profiles, announcements, and channel descriptions

* implementation omissions

* add tab, apply to page editor, and fix something

* componentization

* fix nyaize doesn't working in profile preview

* detach autocomplete instance when unmounted

* fix: mismatched camelCase

* remove unused / unnecessary styles

* update CHANGELOG.md

* fix lint

* remove dump.rdb

* props.richAutocomplete -> autocomplete

* Update packages/frontend/src/scripts/autocomplete.ts

* clarify namings
メンションなども「MFM」に含まれるのか自信がなかったのでrichSyntaxなどとぼかしていましたが、含むようなので変更しました

* tweak

* Update MkFormDialog.vue

* rename

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
update deps

syuilo committed:
2023.12.0-beta.4

samunohito committed:
feat(frontend): 絵文字ピッカーの実装 (#12617)

* 絵文字デッキの作成

* 細かい不備を修正

* fix lint

* fix

* fix CHANGELOG.md

* fix setTimeout -> nextTick

* fix https://github.com/misskey-dev/misskey/pull/12617#issuecomment-1848952862

* fix bug

* fix CHANGELOG.md

* fix CHANGELOG.md

* wip

* Update CHANGELOG.md

* Update CHANGELOG.md

* wip

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
tweak profile.avatar-decoration.dialog.vue

syuilo committed:
New Crowdin updates (#12352)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Czech)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Polish)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Slovak)

* New translations ja-jp.yml (Swedish)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Vietnamese)

* New translations ja-jp.yml (Bengali)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Czech)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Polish)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Slovak)

* New translations ja-jp.yml (Swedish)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Vietnamese)

* New translations ja-jp.yml (Bengali)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Romanian)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (Romanian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Czech)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Greek)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Dutch)

* New translations ja-jp.yml (Polish)

* New translations ja-jp.yml (Portuguese)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Slovak)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Vietnamese)

* New translations ja-jp.yml (Bengali)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Uzbek)

* New translations ja-jp.yml (Lao)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Catalan)

* New translations ja-jp.yml (Catalan)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Chinese Traditional)

1Step621 committed:
fix: contextmenu does not appear when plugin enabled / devMode enabled (#12656)

samunohito committed:
fix(frontend) リアクションの表示位置を微調整 (#12651)

* fix(frontend) リアクションの表示位置を微調整

* fix inline-flex

syuilo committed:
fix(frontend): fix missing localization

YAVIIGI committed:
feat(frontend): MFM のアニメーション要素に delay オプションを追加 (#12659)

* Add delay option

* Update CHANGELOG.md

woxtu committed:
Cleanup unused dependencies (#12660)

syuilo committed:
update deps

samunohito committed:
(dev-mode) devモードの改善 (#12639)

* fix dev-mode

* fix dev-mode

* fix dev-mode

* fix dev-mode

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

hideki0403 committed:
fix: devモードの特定条件下でページが表示されなくなる問題を修正 (#12653)

* fix: devモードでキャッシュクリアするとページが表示されなくなる問題を修正

* fix: localeがnullの場合も最新のlocaleを取得するように

syuilo committed:
enhance: アイコンデコレーションの位置を微調整できるように

woxtu committed:
Remove an unnecessary type assertion (#12666)

anatawa12 committed:
perf: early return users/notes and users/featured-notes if me is blocked by requesting user (#12663)

syuilo committed:
perf(frontend): introduce MkLazy for lazy loading

1Step621 committed:
コード入力ボックスでTabを入力できるように (#12671)

syuilo commented:
vitestでIntersactionObserverが無いって怒られるのはどうしたら良いかしら

tamaina commented:
https://vitest.dev/guide/mocking.html#globals

1Step621 committed:
Fix(frontend): ほとんどのMkTextareaでMFMプレビューが表示されてしまっている問題を修正 (#12672)

* ほとんどのMkTextareaでMFMのプレビューが表示されてしまっている不具合を修正

* refactor

* そもそも #12130 でプロフィールはnyaizeされない仕様にもどっていたらしいので修正

samunohito commented:
なおしました
https://github.com/misskey-dev/misskey/pull/12679

samunohito committed:
fix(frontend): testが落ちるのを修正 (#12679)

FineArchs committed:
Fix: AiScriptの`readline`が不正な値を返すことがある問題を修正 (#12675)

* fix input bug

* Update CHANGELOG.md

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact from 2 to 4 (#12670)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2 to 4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2...v4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

shiosyakeyakini-info committed:
Fix(backend): JSONSchemaに不足しているパラメータを追加 (#12680)

* Fix(backend): JSONSchemaに不足しているパラメータを追加

* nullable:falseに修正

---------

Co-authored-by: sorairo <sorairo@shiosyakeyakini.info>

syuilo committed:
fix(backend): HTTP Digestヘッダのアルゴリズム部分に大文字の"SHA-256"しか使えない

Fix #12678

syuilo committed:
enhance(frontend): tweak user home page

kakkokari-gtyih committed:
enhance(frontend): コードブロックのハイライト機能を利用するには言語を明示的に指定させるように (#12681)

* (enhance) コードブロックのハイライトを使用するには言語指定を求める

* Update changelog

* fix

* typo

syuilo commented:
Get api.json from Misskeyが失敗してるのはなんでだろう

zyoshoka commented:
こちらかと: https://github.com/misskey-dev/misskey/pull/12564#issuecomment-1837496889

anatawa12 commented:
できる対応としては

- 一時的にbaseがmasterの場合にget api.jsonを動かなくする (2023.12.0マージ後 = `generate-api-json`のbase取り込み後みに戻す)\
  (branchesのmasterのところをコメントアウト)
- #12408 より前のように misskeyを起動してcurlで持ってくるように戻す

の２つが思いついています

kakkokari-gtyih committed:
(enhance) コード入力をMkCodeEditorに変更 (#12682)

kakkokari-gtyih committed:
fix(dev): ファイルの読み込みに失敗することがあるのを修正 (#12684)

* (enhance) コード入力をMkCodeEditorに変更

* (fix) devでファイルの読み込みに失敗することがある

* Revert "(enhance) コード入力をMkCodeEditorに変更"

This reverts commit 726d56c3e962680efc5b5a166e2210d09730341f.

samunohito committed:
fix(frontend): 絵文字ピッカー設定ページのボタン文言修正 (#12676)

* fix(frontend): 絵文字ピッカー設定ページのボタン文言修正

* fix

syuilo committed:
clean up

syuilo committed:
feat(frontend): 画面に雪を降らせる機能を復活

syuilo committed:
update deps

syuilo committed:
2023.12.0-beta.5

samunohito commented:
@syuilo 
> Get api.json from Misskey

↑これが失敗することについて、対処する予定でいますか？

（個人的には原因が明確だし、これが失敗するのは無視でいいかと考えています）

syuilo commented:
とりあえず放置しましょう

fruitriin committed:
feature: チャンネル投稿はフォームに色がつくとうれしい (#12686)

syuilo committed:
enhance(frontend): tweak avatar decoration setting ui

fruitriin committed:
fix lint error (#12692)

ikasoba committed:
設定を付けた & 設定自体をフォーマット (#12693)

syuilo committed:
fix(backend): add missing kind definition for admin endpoints to improve security

syuilo committed:
Merge branch 'develop' of https://github.com/misskey-dev/misskey into develop

syuilo committed:
refactor(frontend): use GLSL ES 3.0

syuilo committed:
update deps

syuilo committed:
update mfm-js

zyoshoka committed:
enhance: Unicode 15.0で新たに追加された絵文字をリアクションに使えるように (#12683)

syuilo committed:
Update CHANGELOG.md

1Step621 committed:
Enhance(frontend): Shift+Tabで前の補完候補が選択できるように (#12704)

* Shift+Tabで前の補完候補が選択できるように

* update CHANGELOG.md

1Step621 committed:
Fix: Renoteの判定が間違っているのを修正 (#12706)

* RNと引用RNの判定が間違っているのを修正

* remove dump.rdb

* update CHANGELOG.md

* lint fix

Commit:
enhance: “つながりの公開範囲”がフォロー・フォロワー個別設定できるように (#12702)

* Enhance: “つながりの公開範囲”がフォロー・フォロワー個別設定できるように (#12072)

* refactor: crowdin 編集部分のコミットを打ち消し

https://github.com/misskey-dev/misskey/pull/12702#issuecomment-1859417158

* refactor: オブジェクトの名前修正

https://github.com/misskey-dev/misskey/pull/12702#issuecomment-1859417158

* fix: 設定項目の説明を削除

名称が具体的になって必要なくなったため
https://github.com/misskey-dev/misskey/pull/12702#discussion_r1429932463

1Step621 committed:
Fix(frontend): CWボタンに常に「引用」が出てしまうのを修正 (#12715)

* RNと引用RNの判定が間違っているのを修正

* remove dump.rdb

* update CHANGELOG.md

* lint fix

* fix cw button label

syuilo committed:
New Crowdin updates (#12652)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (Romanian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Catalan)

* New translations ja-jp.yml (Czech)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Greek)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Dutch)

* New translations ja-jp.yml (Norwegian)

* New translations ja-jp.yml (Polish)

* New translations ja-jp.yml (Portuguese)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Slovak)

* New translations ja-jp.yml (Swedish)

* New translations ja-jp.yml (Turkish)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Vietnamese)

* New translations ja-jp.yml (Bengali)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Uzbek)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Korean (Gyeongsang))

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Indonesian)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Spanish)

* New translations ja-jp.yml (Arabic)

* New translations ja-jp.yml (Czech)

* New translations ja-jp.yml (German)

* New translations ja-jp.yml (Italian)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (Polish)

* New translations ja-jp.yml (Portuguese)

* New translations ja-jp.yml (Russian)

* New translations ja-jp.yml (Slovak)

* New translations ja-jp.yml (Ukrainian)

* New translations ja-jp.yml (Chinese Simplified)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (English)

* New translations ja-jp.yml (Vietnamese)

* New translations ja-jp.yml (Bengali)

* New translations ja-jp.yml (Thai)

* New translations ja-jp.yml (Japanese, Kansai)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Chinese Traditional)

GrapeApple0 committed:
fix: 追加絵文字候補辞書の修正(主に国旗) (#12716)

* fix: 絵文字候補辞書の修正(主に国旗)

* 間違って削除してしまった絵文字の復旧

1Step621 committed:
Fix(frontend): CW内などの画像が表示されないことがあるバグを修正 (#12721)

* 画像のアスペクト比をデフォルト以外に設定しているとCW内の画像が表示されないのを修正

* fix

acid-chicken commented on packages/frontend/src/boot/main-boot.ts:
```diff
@@ -19,6 +19,8 @@ import { claimAchievement, claimedAchievements } from '@/scripts/achievements.js
 import { initializeSw } from '@/scripts/initialize-sw.js';
 import { deckStore } from '@/ui/deck/deck-store.js';
+import { emojiPicker } from '@/scripts/emoji-picker.js';
+import { SnowfallEffect } from '@/scripts/snowfall-effect.js';
```
これ main-boot で降ってくるのデカそう

syuilo replied:
snowfall-effect は dynamic import するようにした

saschanaz committed:
chore(QueueProcessorService): show error stack for failures (#12727)

saschanaz committed:
chore(workflows): use postgres 15 everywhere (#12726)

syuilo committed:
enhance: ロールにアサインされたときの通知 (#12607)

* wip

* Update misskey-js.api.md

* Update CHANGELOG.md

* Update RoleService.ts

* Update locales/ja-JP.yml

Co-authored-by: かっこかり <67428053+kakkokari-gtyih@users.noreply.github.com>

* Update UserListService.ts

* Update misskey-js.api.md

* fix (#12724)

---------

Co-authored-by: かっこかり <67428053+kakkokari-gtyih@users.noreply.github.com>
Co-authored-by: おさむのひと <46447427+samunohito@users.noreply.github.com>

syuilo committed:
fix(backend): ロールアサインの通知がある状態でページをリロードすると通知欄に何もでなくなる

Fix #12729

syuilo committed:
perf(frontend): import snowfall-effect dynamically to reduce bundle size

syuilo committed:
Update CHANGELOG.md

anatawa12 committed:
chore: remove hashtag from featured immediately (#12668)

* chore: remove hashtag from featured immediately

* docs(changelog): ハッシュタグのトレンド除外設定が即時に効果を持つように修正

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

woxtu committed:
Remove unused imports (#12730)

dependabot[bot] committed:
chore(deps): bump actions/setup-node from 4.0.0 to 4.0.1 (#12713)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4.0.0...v4.0.1)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

syuilo committed:
New Crowdin updates (#12722)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

syuilo committed:
2023.12.0-beta.6

syuilo commented:
マージするか

samunohito commented:
バックエンドのテストが↓で死んでるので、これを直してからやりたい気持ちもちょっとあります

```
FAIL test/unit/DriveService.ts
  ● Test suite failed to run

    ReferenceError: Cannot access 'RoleService' before initialization

      38 |
      39 | 		private userEntityService: UserEntityService,
    > 40 | 		private idService: IdService,
```

zyoshoka commented on CHANGELOG.md:
```diff
@@ -5,19 +5,116 @@
+## 2023.x.x (unreleased)
+
+### Note
+- Node.js 20.10.0が最小要件になりました
```
#12683 の影響で、アップデート前に絵文字の追加辞書をインストールしていた場合に #12260 が見事に再発するようなので、再インストールを促す文言を changelog に追加しておきたいです（気づくの遅くなって申し訳ないです）。
```suggestion
- Node.js 20.10.0が最小要件になりました
- 絵文字の追加辞書を既にインストールしている場合は、お手数ですが再インストールのほどお願いします
```

syuilo committed:
Update CHANGELOG.md

Co-authored-by: zyoshoka <107108195+zyoshoka@users.noreply.github.com>

syuilo commented:
> バックエンドのテストが↓で死んでるので、これを直してからやりたい気持ちもちょっとあります
> 
> ```
> FAIL test/unit/DriveService.ts
>   ● Test suite failed to run
> 
>     ReferenceError: Cannot access 'RoleService' before initialization
> 
>       38 |
>       39 | 		private userEntityService: UserEntityService,
>     > 40 | 		private idService: IdService,
> ```

直すには循環参照取り除くしかないかも

GrapeApple0 committed:
enhance: api.jsonのレスポンスの内容を実際の内容に合わせる (#12723)

* Create packedAdSchema

* admin/emoji/add

* admin/get-user-ips

* admin/roles/users

* admin/get-index-stats

* admin/accounts/find-by-email

* fix type of admin/ad/list

* federation/stats

* endpoints

* get-online-users-count

* i/2fa/register-key

* i/2fa/key-done

* i/2fa/register

* i/apps

* i/authorized-apps

* i/registry/get-all

* i/registry/get

* i/registry/get-detail

* i/registry/key-with-type

* i/registry/scopes-with-domain

* i/update-email

* i/move

* i/webhooks/create

* fix miss type

* i/webhooks/show

* i/webhooks/list

* flash/create

* roles/users

* server-info

* test

* users/lists/get-memberships

* users/achievements

* fetch-rss

* fetch-external-resources

syuilo commented:
おごごごご

syuilo commented:
テストが動かない問題を解決された方に報奨金を進呈

zyoshoka commented on packages/backend/src/core/RoleService.ts:
```diff
@@ -115,6 +131,10 @@ export class RoleService implements OnApplicationShutdown {
 	}
 
+	async onModuleInit() {
+		this.notificationService = this.moduleRef.get(NotificationService.name);
```
気の所為だったら申し訳ないのですが、これで治りませんかね？
```suggestion
		this.notificationService = this.moduleRef.get('NotificationService');
```

samunohito commented:
多分これで直る…はず
https://github.com/misskey-dev/misskey/pull/12738

samunohito committed:
fix(backend): テストが死んでいるのを修正 (#12738)

* fix test

* fix test

* fix test

* fix test

* fix test

samunohito commented:
あと、これも入れておかないとまずい気が
https://github.com/misskey-dev/misskey/issues/12739

anatawa12 committed:
fix: 引用RNがpure RNとして連合され、pure RNが引用RNとして連合される (#12744)

* fix: quote notes are rendered as pure renote

* fix: filesが指定されてて空配列のときにQuote扱いされる

* chore: isQuoteの仕様をmisc/is-quote.tsと揃える

* docs: is-quote.tsの方にNoteCreateService.isQuoteのことを書いて更新忘れを防ぐ

zyoshoka committed:
fix(i18n): ロールが付与された際の通知のローカライゼーションが一部欠けているのを修正 (#12745)

syuilo committed:
New Crowdin updates (#12732)

* New translations ja-jp.yml (Chinese Traditional)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Catalan)

* New translations ja-jp.yml (Catalan)

* New translations ja-jp.yml (English)

syuilo commented:
マージするぞ

syuilo commented:
Misskey Hub Nextが今日リリースできるかもしれないからそれまで待機(https://github.com/misskey-dev/misskey/pull/12699 マージしてからリリースしたい)

syuilo committed:
fix type

kakkokari-gtyih committed:
[Hub Next] Misskey Hubのリンクを変更 (#12699)

* misskey hub のリンクを変更

* misskey-hub.net -> assets.misskey-hub.net
Related to misskey-dev/misskey-hub-next#57

syuilo commented:
> Misskey Hub Nextが今日リリースできるかもしれないからそれまで待機(#12699 マージしてからリリースしたい)

done

ikasoba committed:
enhance: チャンネルに新規の投稿がある場合にバッジを表示させる (#12690)

* 多分できたかも

* 不要なpropsを削除

* 不要なimportを削除

* 縁を付けた

* 枠線の位置を端に寄せた

* やっぱり内側へ寄せることにした

* できたかも

* 修正

* 修正

* クラスにまとめた

* 微調整

* 直せたかも

* importを付け足し

* 多分できたかも

* Update channel.vue

* Update MkMenu.vue

* Update channel.vue

* Update CHANGELOG.md

---------

Co-authored-by: syuilo <Syuilotan@yahoo.co.jp>

syuilo committed:
Update CHANGELOG.md

syuilo committed:
2023.12.0

syuilo commented:
マージします

syuilo commented:
一人以上からapprove貰えたら

zyoshoka commented on CHANGELOG.md:
```diff
@@ -5,19 +5,125 @@
 
 ### Client
--
+- Fix: ページ一覧ページの表示がモバイル環境において崩れているのを修正
+- Fix: MFMでルビの中のテキストがnyaizeされない問題を修正
```
これ直したほうが良さそうです

syuilo committed:
Update CHANGELOG.md

syuilo commented:
マージするか

syuilo commented:
うおおおおおおお

Candinya committed:
fix: email verify enable logic (#12743)

GrapeApple0 committed:
feat: 登録を拒否するメールアドレスのドメインを手動で設定できるように (#12740)

* feat: 使い捨てアドレスのドメインを手動で設定できるように

* Update CHANGELOG.md

* disposableEmailDomains -> bannedEmailDomains

* isBlockedHostを使うように

syuilo committed:
Update CHANGELOG.md

samunohito committed:
fix: 2023.12.0のNoteの一部文言を修正 (#12754)

syuilo committed:
lint

syuilo commented:
マージするぞ

samunohito commented:
ﾁｮｯﾄﾏｯﾃ…

samunohito commented on CHANGELOG.md:
```diff
@@ -5,19 +5,127 @@
+- Fix: 共有機能をサポートしていないブラウザの場合は共有ボタンを非表示にする #11305
+- Fix: 通知のグルーピング設定を変更してもリロードされるまで表示が変わらない問題を修正 #12470
+- Fix: 長い名前のチャンネルにおける投稿フォームの表示が崩れる問題を修正
+- Fix: セキュリティ向上のためAiScriptの`Mk:apiExternal`を無効化
```
> Fix: セキュリティ向上のためAiScriptの`Mk:apiExternal`を無効化

この部分

kakkokari-gtyih replied:
これ、裏でSecurity Advisoryが出ているので、リリース後に対応したほうが良いかも

samunohito committed:
fix: .npmrcによりpackage.json記載のnodeバージョンに満たない場合はビルドに失敗するようにする (#12755)

nafu-at committed:
chore(frontend): API設定項目の名前をきちんとサービスの名前で表記する (#12753)

tkmrgit commented:
>Enhance: Meilisearchを有効にした検索で、ユーザーのミュートやブロックを考慮するように
#10844 の対応ということでOK？

syuilo committed:
Update CHANGELOG.md

syuilo committed:
New Crowdin updates (#12748)

* New translations ja-jp.yml (Korean)

* New translations ja-jp.yml (French)

* New translations ja-jp.yml (Chinese Traditional)

syuilo commented:
マージするぞ

syuilo commented:
うおおおおおおお

samunohito commented on packages/backend/src/core/NoteCreateService.ts:
```diff
@@ -292,7 +293,7 @@ export class NoteCreateService implements OnApplicationShutdown {
 
 		// Check blocking
-		if (data.renote && data.text == null && data.poll == null && (data.files == null || data.files.length === 0)) {
+		if (this.isQuote(data)) {
```
ここ条件裏返ってないですか？

anatawa12 replied:
ホントだ。直します。

syuilo replied:
ほんまや

syuilo replied:
直しました 🙏🏻

syuilo committed:
fix(backend): renote判定がおかしい

syuilo committed:
Revert "fix(backend): renote判定がおかしい"

This reverts commit f43599552fb5764aa3121b083e441d3946c72cd8.

anatawa12 committed:
fix: ブロックされててもpure RNできる (#12758)

なぜかわからないけど元々Quoteはできるようなのでそれに戻しました

anatawa12 commented:
コメント指定でもcloseされちゃうのか。ごめんなさい。Reopenお願いします

syuilo commented:
マージをやっていく

syuilo commented:
うおおおおおおお

syuilo commented:
20時にマージするか

acid-chicken commented:
エントリモジュールに絵文字辞書入ってたのって前からだっけ？

syuilo commented:
> エントリモジュールに絵文字辞書入ってたのって前からだっけ？

だった気がする

syuilo committed:
Update CHANGELOG.md

syuilo commented:
🤯

syuilo commented:
🙏🏻🙏🏻🙏🏻🙏🏻🙏🏻
